src/main/java/cuchaz/enigma/command/MapSpecializedMethodsCommand.java

```diff
@@ -0,0 +1,67 @@
+package cuchaz.enigma.command;
+
+import cuchaz.enigma.ProgressListener;
+import cuchaz.enigma.analysis.ClassCache;
+import cuchaz.enigma.analysis.index.BridgeMethodIndex;
+import cuchaz.enigma.analysis.index.JarIndex;
+import cuchaz.enigma.throwables.MappingParseException;
+import cuchaz.enigma.translation.MappingTranslator;
+import cuchaz.enigma.translation.Translator;
+import cuchaz.enigma.translation.mapping.*;
+import cuchaz.enigma.translation.mapping.tree.EntryTree;
+import cuchaz.enigma.translation.mapping.tree.EntryTreeNode;
+import cuchaz.enigma.translation.mapping.tree.HashEntryTree;
+import cuchaz.enigma.translation.representation.entry.MethodEntry;
+import cuchaz.enigma.utils.Utils;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+
+public class MapSpecializedMethodsCommand extends Command {
+    public MapSpecializedMethodsCommand() {
+        super("map-specialized-methods");
+    }
+
+    @Override
+    public String getUsage() {
+        return "<jar> <source-format> <source> <result-format> <result>";
+    }
+
+    @Override
+    public boolean isValidArgument(int length) {
+        return length == 5;
+    }
+
+    @Override
+    public void run(String... args) throws IOException, MappingParseException {
+        MappingSaveParameters saveParameters = new MappingSaveParameters(MappingFileNameFormat.BY_DEOBF);
+        EntryTree<EntryMapping> source = MappingCommandsUtil.read(args[1], Paths.get(args[2]), saveParameters);
+        EntryTree<EntryMapping> result = new HashEntryTree<>();
+        Path jar = Paths.get(args[0]);
+        ClassCache classCache = ClassCache.of(jar);
+        JarIndex jarIndex = classCache.index(ProgressListener.none());
+        BridgeMethodIndex bridgeMethodIndex = jarIndex.getBridgeMethodIndex();
+        Translator translator = new MappingTranslator(source, jarIndex.getEntryResolver());
+
+        // Copy all non-specialized methods
+        for (EntryTreeNode<EntryMapping> node : source) {
+            if (!(node.getEntry() instanceof MethodEntry) || !bridgeMethodIndex.isSpecializedMethod((MethodEntry) node.getEntry())) {
+                result.insert(node.getEntry(), node.getValue());
+            }
+        }
+
+        // Add correct mappings for specialized methods
+        for (Map.Entry<MethodEntry, MethodEntry> entry : bridgeMethodIndex.getBridgeToSpecialized().entrySet()) {
```

Comment:
I always prefer to use a `foreach((k,v) -> {}` lambda for maps but many people prefer it this way so I guess It's just fine

Reply:
lambda one is costly and hard for jvm to optimize.

Reply:
It is super insignificant. One object allocation.

Reply:
Sure, in hot loops it can be a problem, but this is very much not time critical.